### ISSUE TITLE
Signal names starting with an underscore after a dash are not mapped correctly

### ISSUE THREAD

**Issue by stefanohrmann** [starfederation/datastar#404]
According to the documentation of the [Attribute Plugins](https://data-star.dev/reference/attribute_plugins) the following definition of a signal should result in the same signal:

```html
<pre data-signals-_foo="1"></pre>
<pre data-signals="{ _foo: 2 }"></pre>
```

But instead two different signals are created:

```
{
  "-_foo": 1,
  "_foo": "2",
}
```

The same behaviour occurs with `data-indicator-*`. It is possible to see both behaviours with the following example:

```html
<!DOCTYPE html>
<html lang="en">

<head>
    <script type="module" src="https://cdn.jsdelivr.net/gh/starfederation/datastar@develop/bundles/datastar.js"></script>
</head>

<body>
    <pre data-indicator-_loading></pre>
    <pre data-indicator="_loading"></pre>
    <pre data-signals-_foo="1"></pre>
    <pre data-signals="{ _foo: 2 }"></pre>
    <pre data-text="ctx.signals.JSON()"></pre>
</body>

</html>
```

**Comment by bencroker** [comment]
Thanks for the detailed report! I’ve raised a PR at #407.

**Comment by bencroker** [comment]
Fixed by #407.
